akka-http2-support/src/main/scala/akka/http/scaladsl/Http2.scala

```diff
@@ -84,6 +88,65 @@ final class Http2Ext(private val config: Config)(implicit val system: ActorSyste
       }.to(Sink.ignore).run()
   }
 
+  private def handleUpgradeRequests(
+    handler:     HttpRequest => Future[HttpResponse],
+    settings:    ServerSettings,
+    parallelism: Int,
+    log:         LoggingAdapter
+  ): HttpRequest => Future[HttpResponse] = { req =>
+    req.header[Upgrade] match {
+      case Some(upgrade) if upgrade.protocols.exists(_.name equalsIgnoreCase "h2c") =>
```

Comment:
Feels like an if-else rather than a patmatch with catchall to me?

`if (req.header[Upgrade].exists(_.name.equalsIgnoreCase("h2x"))) {} else {}`

Maybe worth also caching that lambda since it is evaluated for each request?

Reply:
it's a little more tricky because the `exists` is on the list of protocols, not on the option... good point about the lambda though, will try to rewrite

Reply:
Hmm actually I struggle coming up with a solution that is meaningfully better, could you suggest something?

Reply:
Nah, leave it be in that case.